pandas/core/frame.py

```diff
@@ -4166,8 +4173,9 @@ def _set_item(self, key, value) -> None:
                 existing_piece = self[key]
                 if isinstance(existing_piece, DataFrame):
                     value = np.tile(value, (len(existing_piece.columns), 1)).T
+                    refs = None
```

Comment:
I was wondering if this would still be a view in case we are only setting a single column (i.e. `len(existing_piece.columns)` is equal to 1). But it seems that also in that case `np.tile` consistently returns a copy, so that should be fine. 
(it might still be an opportunity for an optimization, although for a corner case)